src/plugin/helpers/validation.ts

```diff
@@ -19,6 +19,82 @@ export const createValidator = (
   return errors;
 };
 
+type ImageAsset = {
+  fields: {
+    width: number;
+    height: number;
+  };
+};
+
+const hasOwnProperty = <X extends Record<string, unknown>, Y extends string>(
+  obj: X,
+  prop: Y
+): obj is X & Record<Y, unknown> => {
+  return Object.hasOwnProperty.call(obj, prop);
+};
+
+const isRecord = (value: unknown): value is Record<string, unknown> => {
+  return typeof value === "object" && value !== null;
+};
+
+const isValidImageAsset = (
```

Comment:
I would recommend moving this logic into a file located within the image element folder, as it is only likely to be useful for that particular element. When the elements are split out, this divide should become clearer.

Reply:
Sounds very sensible - will do so.